backends/rapidpro/backend.go

```diff
@@ -536,11 +542,24 @@ func (b *backend) WriteStatusUpdate(ctx context.Context, status courier.StatusUp
 		}
 	}
 
-	// if we have an id and are marking an outgoing msg as errored, then clear our sent flag
-	if status.ID() != courier.NilMsgID && status.Status() == courier.MsgStatusErrored {
-		err := b.ClearMsgSent(ctx, status.ID())
-		if err != nil {
-			logrus.WithError(err).WithField("msg", status.ID()).Error("error clearing sent flags")
+	if status.ID() != courier.NilMsgID {
+		// this is a message we've just sent and were given an external id for
+		if status.ExternalID() != "" {
+			rc := b.redisPool.Get()
+			defer rc.Close()
+
+			err := b.sentExternalIDs.Set(rc, fmt.Sprintf("%d|%s", su.ChannelID_, su.ExternalID_), fmt.Sprintf("%d", status.ID()))
```

Comment:
we should update the other hashes to use channel id instead of uuid.. there's no advantage to using uuid here and it's more redis storage